apps/frontend/src/hooks/algolia.ts

```diff
@@ -1,26 +1,46 @@
 import { useEffect, useState } from 'react';
-import algoliasearch from 'algoliasearch/lite';
 import { useCurrentUser } from '@asap-hub/react-context';
+import {
+  algoliasearchLite,
+  SearchClientLite,
+  SearchIndex,
+  ResearchOutputSearchIndex,
+} from '@asap-hub/algolia';
 import { User } from '@asap-hub/auth';
 
 import { ALGOLIA_APP_ID, ALGOLIA_INDEX } from '../config';
 
+export type AlgoliaHook = {
+  client: SearchClientLite;
+  index: {
+    researchOutput: ResearchOutputSearchIndex;
+  };
+};
+
 export const useAlgolia = () => {
-  const initAlgolia = (user: User | null) => {
+  const initAlgolia = (user: User | null): AlgoliaHook => {
     if (!user) {
       throw new Error('Algolia unavailable while not logged in');
     }
-    const client = algoliasearch(ALGOLIA_APP_ID, user.algoliaApiKey);
-    const index = client.initIndex(ALGOLIA_INDEX);
+
+    const client = algoliasearchLite(ALGOLIA_APP_ID, user.algoliaApiKey);
+    const researchOutput = new ResearchOutputSearchIndex(
```

Comment:
I am a bit confused on what does this do over just having the `client.initIndex(ALGOLIA_INDEX)` ? is it for the types?

Reply:
pretty much :D but i think type safety is quite important so

Reply:
we can pass the response type to the client search method, which was what we were doing. the only thing saying which response type we get is the facet filter we send, so I don't see this giving us more type safety. We just would have a  few clients instantiated pointing to the same index.